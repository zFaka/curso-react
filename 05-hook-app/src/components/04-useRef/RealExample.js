import React, {useState} from 'react';
import {MultipleCustomHooks} from '../03-examples/MultipleCustomHooks';
import '../02-useEffect/effects.css' ;

const RealExample = () => {

  const [show, setShow] = useState(false);
  const handleHider = () => {setShow(!show)};

  return (
    <>
      <h1>RealExample</h1> 
      <hr/>
      {show && <MultipleCustomHooks/>}
      <hr/>
      <button 
        className='btn btn-outline-primary'
        onClick={handleHider}
      >
        Show/Hide
      </button>
    </>
  )}

export default RealExample;
