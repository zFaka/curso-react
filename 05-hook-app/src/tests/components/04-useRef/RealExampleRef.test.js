import {shallow} from "enzyme"
import RealExample from "../../../components/04-useRef/RealExample"

describe('Test RealExample', () => {


  const wrapper = shallow(<RealExample/>);



  test('Should load Component', () => {

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('MultipleCustomHooks').exists())
      .toBe(false);
  })



  test('Should show MultipleCustomHooks', () => {

    wrapper.find('button').simulate('click');

    expect(wrapper.find('MultipleCustomHooks').exists())
      .toBe(true);
  })
})
