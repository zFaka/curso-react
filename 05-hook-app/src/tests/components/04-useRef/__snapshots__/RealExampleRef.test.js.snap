// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RealExample Should load Component 1`] = `
<Fragment>
  <h1>
    RealExample
  </h1>
  <hr />
  <hr />
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;
